Upgrade golangci-lint to 1.20

golangci-lint 1.20 reported the following error and failed the build

pkg/procmanager/procmanager.go:111: G204: Subprocess launched with function call as argument or cmd arguments (gosec)
        cmd := exec.Command(args[0], args[1:]...)
Makefile:71: recipe for target 'lint' failed

--- a/cmd/acb/commands/exec/exec.go
+++ b/cmd/acb/commands/exec/exec.go
@@ -150,7 +150,7 @@ var Command = cli.Command{
 			taskFile = defaultTaskFile
 		}
 
-		ctx := gocontext.WithValue(gocontext.Background(), "debug", debug)
+		ctx := gocontext.Background()
 		pm := procmanager.NewProcManager(dryRun)
 
 		if homevol == "" {
@@ -228,7 +228,7 @@ var Command = cli.Command{
 			if aliasErr != nil {
 				return errors.Wrap(aliasErr, "unable to search/replace aliases in task")
 			}
-			if ctx.Value("debug").(bool) {
+			if debug {
 				log.Printf("Processed task before rendering data:\n%s", processedTask)
 			}
 			// update the template.Data
@@ -239,7 +239,7 @@ var Command = cli.Command{
 		if err != nil {
 			return errors.Wrap(err, "unable to render task")
 		}
-		if ctx.Value("debug").(bool) {
+		if debug {
 			log.Printf("Rendered template:\n%s", rendered)
 		}
 

--- a/pkg/procmanager/procmanager.go
+++ b/pkg/procmanager/procmanager.go
@@ -130,7 +130,7 @@ func (pm *ProcManager) Run(
 		return nil
 	}
 
-	cmd := exec.Command(args[0], args[1:]...)
+	cmd := exec.Command(args[0], args[1:]...) //#nosec G204
 	if cmdDir != "" {
 		cmd.Dir = cmdDir
 	}

--- a/scripts/setup/dev_setup
+++ b/scripts/setup/dev_setup
@@ -3,4 +3,4 @@
 set -e -o pipefail
 
 # binary will be $(go env GOPATH)/bin/golangci-lint
-curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.20.1/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.20.1
+curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.27.0/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.27.0

--- a/scripts/setup/dev_setup
+++ b/scripts/setup/dev_setup
@@ -3,4 +3,4 @@
 set -e -o pipefail
 
 # binary will be $(go env GOPATH)/bin/golangci-lint
-curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.17.1/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.17.1
+curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.20.1/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.20.1

